tests/modules/goat/taxonsearch/main.nf

```diff
@@ -0,0 +1,31 @@
+#!/usr/bin/env nextflow
+
+nextflow.enable.dsl = 2
+
+include { GOAT_TAXONSEARCH } from '../../../../modules/goat/taxonsearch/main.nf'
+
+//
+// Test with genus name (Canis)
+//
+workflow test_goat_taxonsearch_genus_name {
+
+    input = [
+      [ id:'test_genus_name' ], // meta map
+      taxon = 'Canis',
+      []
+  ]
+    GOAT_TAXONSEARCH ( input )
+}
+
+//
+// Test with genus (Drosophila, fruit flies) using NCBI taxonomy ID
+//
+workflow test_goat_taxonsearch_genus_id {
```

Comment:
Could you also add a test with a taxa file?

Reply:
I guess is possible, I would need to create a test file in the [nf-core test datasets repo](https://github.com/nf-core/test-datasets) and then fetch the file from there, sorry for the late response.

Reply:
No problem! Are these taxa files big? Maybe post in the `test-data` slack channel before adding it. And see if it's okay to add :)

Reply:
They only contain the NCBI tax IDs or binomial names to search (one per line,  limited to 500 lines max.), so it should be small.

Reply:
That should be fine then, I think

Reply:
Hi, I just added the .txt file that will be used for the test, here is the PR: [test file for goat/taxonsearch module #590](https://github.com/nf-core/test-datasets/pull/590), please check it before merging it, I will add the test to the module today.

Reply:
Done, I just added a tests using an input taxa file.

Reply:
> Could you also add a test with a taxa file?

Done, I just added a test using an input taxa file